framework/src/libtorch/userobjects/TorchScriptUserObject.C

```diff
@@ -0,0 +1,57 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifdef LIBTORCH_ENABLED
+
+#include "TorchScriptUserObject.h"
+
+registerMooseObject("MooseApp", TorchScriptUserObject);
+
+InputParameters
+TorchScriptUserObject::validParams()
+{
+  InputParameters params = GeneralUserObject::validParams();
+  params.addClassDescription("User-facing object which loads a torch script module.");
+  params.addRequiredParam<std::string>("filename",
+                                       "The file name which contains the torch script module.");
+  params.declareControllable("filename");
+  params.addParam<bool>(
+      "load_during_construction",
+      false,
+      "If we want to load this neural network while we are constructing this object.");
+  return params;
+}
+
+TorchScriptUserObject::TorchScriptUserObject(const InputParameters & parameters)
+  : GeneralUserObject(parameters),
+    _filename(getParam<std::string>("filename")),
+    _torchscript_module(getParam<bool>("load_during_construction")
+                            ? std::make_shared<Moose::TorchScriptModule>(_filename)
+                            : nullptr)
```

Comment:
What's the downside of always loading the torch script at construction time? I think if the user uses this object, it is implied that the script module _will_ be used eventually.

Reply:
Hmmm good point, I will just assume there is already a neural net when we set up the problem. I am thinking about peple using this together with the webserver interface and saving before the start of it. But I assume it is pretty easy to save at the very beginning of the python run before they fire up the moose run.

Reply:
I'm fine with a boolean knob now that I understand the intended use case.